SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp

```diff
@@ -0,0 +1,412 @@
+        m_display_order.emplace_back("Losses");
+        m_display_order.emplace_back("Items matched");
+        m_display_order.emplace_back("Errors", true);
+    }
+    std::atomic<uint64_t>& tournaments;
+    std::atomic<uint64_t>& battles;
+    std::atomic<uint64_t>& losses;
+    std::atomic<uint64_t>& matches;
+    std::atomic<uint64_t>& errors;
+};
+std::unique_ptr<StatsTracker> TournamentFarmer_Descriptor::make_stats() const {
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+TournamentFarmer::TournamentFarmer()
+    : NUM_ROUNDS(
+          "<b>Number of Tournaments to run:",
+          LockWhileRunning::UNLOCKED,
+          100, 0
+          )
+    , TRY_TO_TERASTILLIZE(
+          "<b>Use Terastillization:</b><br>Tera at the start of battle. Will take longer but may be worth the attack boost.",
+          LockWhileRunning::UNLOCKED,
+          false
+          )
+    , SAVE_NUM_ROUNDS(
+          "<b>Save every this many tournaments:</b><br>Zero disables saving. Will save win or lose.",
+          LockWhileRunning::UNLOCKED,
+          0, 0
+          )
+    , GO_HOME_WHEN_DONE(false)
+    , LANGUAGE(
+          "<b>Game Language:</b><br>The language is needed to read the prizes.",
+          TournamentPrizeNameReader::instance().languages(),
+          LockWhileRunning::LOCKED,
+          true
+          )
+    , TARGET_ITEMS("<b>Items:</b>")
+    , NOTIFICATION_PRIZE_MATCH("Matching Prize", true, false, ImageAttachmentMode::JPG, { "Notifs" })
+    , NOTIFICATION_STATUS_UPDATE("Status Update", true, false, std::chrono::seconds(3600))
+    , NOTIFICATIONS({
+        &NOTIFICATION_PRIZE_MATCH,
+        &NOTIFICATION_STATUS_UPDATE,
+        &NOTIFICATION_PROGRAM_FINISH,
+        &NOTIFICATION_ERROR_FATAL,
+        })
+{
+    PA_ADD_OPTION(NUM_ROUNDS);
+    PA_ADD_OPTION(TRY_TO_TERASTILLIZE);
+    PA_ADD_OPTION(SAVE_NUM_ROUNDS);
+    PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+    PA_ADD_OPTION(LANGUAGE);
+    PA_ADD_OPTION(TARGET_ITEMS);
+    PA_ADD_OPTION(NOTIFICATIONS);
+}
+
+
+//Handle a single battle by mashing A until a black screen (end of battle) is detected
+void TournamentFarmer::run_battle(SingleSwitchProgramEnvironment& env, BotBaseContext& context) {
+    TournamentFarmer_Descriptor::Stats& stats = env.current_stats<TournamentFarmer_Descriptor::Stats>();
+    //Assuming the player has a charged orb
+    if (TRY_TO_TERASTILLIZE) {
+        //Open move menu
+        pbf_press_button(context, BUTTON_A, 10, 50);
+        pbf_wait(context, 100);
+        context.wait_for_all_requests();
+
+        pbf_press_button(context, BUTTON_R, 20, 50);
+    }
+
+    //Mash A until battle finished
+    BlackScreenWatcher end_of_battle(COLOR_RED, { 0.2, 0.2, 0.6, 0.6 });
+    int ret_black = run_until(
+        env.console, context,
+        [](BotBaseContext& context) {
+            pbf_mash_button(context, BUTTON_A, 30000);
+            pbf_mash_button(context, BUTTON_A, 7500); //5 minutes should be more than enough for one battle
```

Comment:
This kind of A-mashing over such a long duration is a bit risky. If the game crashes and there's an update available, it will run the update.

A better approach here is to search for check for the battle menu, and only mash A for a couple seconds at a time when you see the battle menu. IOW, add `NormalBattleMenuDetector` as an additional callback here to tell you when you press A.